### PR TITLE
expression: implement vectorized evaluation for 'builtinTimeFormatSig'

### PR DESCRIPTION
**Pull request by tsthght**
### What problem does this PR solve? 
Implement vectorized evaluation for builtinTimeFormatSig, for #12106

### What is changed and how it works?
```
BenchmarkVectorizedBuiltinTimeFunc/builtinTimeFormatSig-VecBuiltinFunc-8         	   10000	    101142 ns/op   33347 B/op	    1224 allocs/op
BenchmarkVectorizedBuiltinTimeFunc/builtinTimeFormatSig-NonVecBuiltinFunc-8      	   10000	    123739 ns/op   33347 B/op	    1224 allocs/op
```

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: expression/bench_test.go [hunk 1]
```diff
@@ -18,6 +18,7 @@ package expression
 import (
 	"flag"
 	"fmt"
+	"github.com/pingcap/tidb/util/math"
```
Comment (qw4990) [inline expression/bench_test.go]: Please format this.

File: expression/builtin_time_vec.go [hunk 2]
```diff
@@ -189,11 +189,43 @@ func (b *builtinWeekDaySig) vecEvalInt(input *chunk.Chunk, result *chunk.Column)
...
+	result.ReserveString(n)
+	for i := 0; i < n; i++ {
+		if buf.IsNull(i) || buf1.IsNull(i) {
+			result.AppendNull()
+			continue
+		}
+		res, err := b.formatTime(b.ctx, buf.GetDuration(i, 0), buf1.GetString(i))
+		if err != nil {
+			result.AppendNull()
+			continue
```
Comment (qw4990) [inline expression/builtin_time_vec.go]: It seems we should throw this error here like the row-based implementation.

File: expression/bench_test.go (hunk 1 above)
Comment (tsthght) [reply expression/bench_test.go]: ok

File: expression/builtin_time_vec.go (hunk 2 above)
Comment (tsthght) [reply expression/builtin_time_vec.go]: ok

File: expression/builtin_time_vec.go [hunk 3]
```diff
@@ -189,11 +189,42 @@ func (b *builtinWeekDaySig) vecEvalInt(input *chunk.Chunk, result *chunk.Column)
...
+	buf, err := b.bufAllocator.get(types.ETDuration, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf)
+	if err := b.args[0].VecEvalDuration(b.ctx, input, buf); err != nil {
+		return err
+	}
+
+	buf1, err1 := b.bufAllocator.get(types.ETString, n)
```
Comment (qw4990) [inline expression/builtin_time_vec.go]:
```suggestion
	buf1, err := b.bufAllocator.get(types.ETString, n)
```

**Pull request merged**